docs/android/features/performance-instrumentation.md

```diff
@@ -0,0 +1,210 @@
+
+On Android version 6.x or earlier, the trace will start when the Embrace SDK starts, which is recommended to be the first line of the `onCreate()` method of the app's `Application` object.
+
+Warm startups generate a trace with a root span called `emb-warm-time-to-initial-display`. This trace begins at our estimate of when the designated Activity creation begins. It will end when the first frame of the designated Activity has been fully rendered.
+
+For Android version 9 or earlier, both types of app startup traces will end when the designated Activity reaches the `RESUMED` state of its lifecycle.
+
+### Configuration
+
+App startup traces can be augmented in the following ways to add additional data particular to your app:
+
+- Custom attributes
+- Custom child spans
+- Application object creation completion
+- Ignore splash or trampoline Activities on startup
+
+Note: Be mindful of when these customization methods are invoked. Doing so on a background thread or at a time when it is not certain whether app startup has completed may result in the customization not being recorded.
+
+#### Custom attributes
+
+Custom span attributes can be added to the root span of the trace by using the `addStartupTraceAttribute()` method before startup has completed.
+
+#### Custom child spans
+
+Custom child spans can be added to the root span of the trace during app startup by using the `addStartupTraceAttribute()` method. Parameters for the name, start time, and end time must be specified, while optional parameters for attributes, span events, and error code may also be specified to further customize the span.
+
+To ensure the timestamps of the custom spans are in sync with the timestamps of the other spans in the trace, use the `getSdkCurrentTimeMs()` method to obtain it from the same clock instance the SDK uses. This instance is locked after the SDK starts up and will not change even if the system clock changes.
+
+#### Application object create completion notification
+
+It's difficult to programmatically determine precisely when the app's application object has been created, but the `applicationInitEnd()` method can be called to notify the SDK of when this happens, which will allow it to more accurately assess whether an app startup is cold or warm.
+
+#### Bypassing interstitial activities during app startup
+
+For any interstitial Activities that are opened before the actual Activity whose load completes the launch of the app, annotate them with the `@StartupActivity` annotation so that they will be skipped for the purposes of app startup tracking.
+
+### Code Example
+
+The following sample `Application` will add a custom span and attribute to the app startup trace. It will also inform the SDK of when the application object has finished beign created.
```

Comment:
```suggestion
The following sample `Application` will add a custom span and attribute to the app startup trace. It will also inform the SDK of when the application object has finished being created.
```